pennylane/measurements.py

```diff
@@ -288,6 +291,85 @@ def shape(self, device=None):
             + f"{self.return_type}."
         )
 
+    @functools.lru_cache()
+    def _shape_new(self, device=None):
+        """The expected output shape of the MeasurementProcess.
+
+        Note that the output shape is dependent on the device when:
+
+        * The ``return_type`` is either ``Probability``, ``State`` (from :func:`.state`) or
```

Comment:
Potentially also can be entropy, mutual info as well.

Reply:
In the case of entropy and mutual info, the output shape is always scalar and independent of the device.